lib/Parser/Parser.yy

```diff
@@ -50,155 +50,604 @@
+    { std::cerr << "Parsed enum declaration" << std::endl; }
     ;
 
-struct_members
-    : /* empty */
-    | struct_members struct_member
+enum_body:
+      lBrace enum_variant_list_opt rBrace
     ;
 
-struct_member
-    : var_decl
+enum_variant_list_opt:
+      /* empty */
+    | enum_variant_list
     ;
 
-compound_stmt
-    : lBrace statements rBrace
+enum_variant_list:
+      enum_variant
+    | enum_variant_list comma enum_variant
+    | enum_variant_list comma
     ;
 
-statements
-    : /* empty */
-    | statements statement
+enum_variant:
+      ident
+    | ident equal expression
     ;
 
-statement
-    : expr_stmt
-    | compound_stmt
+typealias_declaration:
+      attributes typealiasKw ident template_definition_opt equal type semi
+    { std::cerr << "Parsed typealias declaration" << std::endl; }
+    ;
+
+/*--------------------------------*/
+/* Function declarations          */
+/*--------------------------------*/
+
+function_declaration:
+      attributes funcKw ident template_definition_opt function_signature function_body
+    { std::cerr << "Parsed function declaration" << std::endl; }
+    ;
+
+function_signature:
+      lParen parameter_list_opt rParen arrow type
+    | lParen parameter_list_opt rParen
+    ;
+
+parameter_list_opt:
+      /* empty */
+    | parameter_list
+    ;
+
+parameter_list:
+      parameter
+    | parameter_list comma parameter
+    | parameter_list comma
+    ;
+
+parameter:
+      ident colon type
+    | ident colon type equal expression
+    ;
+
+function_body:
+      block
+    | semi
+    ;
+
+block:
+      lBrace statement_list rBrace
+    ;
+
+statement_list:
+      /* empty */
+    | statement_list statement
+    ;
+
+statement:
+      block
+    | expression_stmt semi
+    | var_stmt
+    | let_stmt
+    | return_stmt
     | if_stmt
     | while_stmt
-    | return_stmt
+    | for_stmt
+    | break_stmt
+    | continue_stmt
+    | assignment_stmt
+    | semi
+    ;
+
+expression_stmt:
+      conditional_expression_stmt
+    ;
+
+conditional_expression_stmt:
+  logical_or_expression_stmt
+    | logical_or_expression_stmt question expression colon conditional_expression_stmt %prec TERNARY
+      { std::cerr << "Parsed ternary expression" << std::endl; }
+    ;
+
+logical_or_expression_stmt:
+  logical_or_expression_stmt orOp logical_and_expression_stmt
+      { std::cerr << "Parsed logical or" << std::endl; }
+    | logical_and_expression_stmt
+
+logical_and_expression_stmt:
+  logical_and_expression_stmt andOp conditional_expression_stmt
+      { std::cerr << "Parsed logical and" << std::endl; }
+    | function_call_stmt
+    ;
+
+function_call_stmt:
+  primary_expression
+    | function_call_stmt function_template_arguments lParen argument_list_opt rParen %prec POSTFIX
+      { std::cerr << "Parsed function call with template arguments" << std::endl; }
+    | function_call_stmt lParen argument_list_opt rParen %prec POSTFIX
+      { std::cerr << "Parsed function call" << std::endl; }
```

Comment:
why not have one same rule, and add an empty production to `function_template_arguments`?